sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -86,6 +71,78 @@ case class Mode(
     buffer
   }
 
+  private def recursivelyGetBufferForArrayType(
+      a: ArrayType,
+      data: ArrayData): Seq[Any] = {
+    (0 until data.numElements()).map { i =>
+      data.get(i, a.elementType) match {
+        case k: UTF8String if a.elementType.isInstanceOf[StringType] &&
+          !a.elementType.asInstanceOf[StringType].supportsBinaryEquality
+          => CollationFactory.getCollationKey(k, a.elementType.asInstanceOf[StringType].collationId)
```

Comment:
I think we can use `UnsafeRowUtils` here as well

Reply:
Done